aws/resource_aws_iam_user_login_profile.go

```diff
@@ -59,27 +59,14 @@ func resourceAwsIamUserLoginProfile() *schema.Resource {
 // characters that are likely to satisfy any possible AWS password policy
 // (given sufficient length).
 func generatePassword(length int) string {
-	charsets := []string{
-		"abcdefghijklmnopqrstuvwxyz",
-		"ABCDEFGHIJKLMNOPQRSTUVWXYZ",
-		"012346789",
-		"!@#$%^&*()_+-=[]{}|'",
-	}
+	charset := `abcdefghijklmnopqrstuvwxyz
+ABCDEFGHIJKLMNOPQRSTUVWXYZ
+012346789
+!@#$%^&*()_+-=[]{}|'`
 
-	// Use all character sets
-	components := make(map[int]byte, length)
-	for i := 0; i < length; i++ {
-		charset := charsets[i%len(charsets)]
-		components[i] = charset[rand.Int(len(charset))]
-	}
-
-	// Randomise the ordering so we don't end up with a predictable
-	// lower case, upper case, numeric, symbol pattern
 	result := make([]byte, length)
-	i := 0
-	for _, b := range components {
-		result[i] = b
-		i = i + 1
+	for i := 0; i < length; i++ {
+		result[i] = charset[rand.Int(len(charset))]
```

Comment:
It's technically possible to have a password now that is exactly one character, if somehow `rand.Int()` always returns 0. I am not sure if there is a `crypto/rand` alternative to `rand.Perm()`, which returns a random permutation.

Reply:
If by “exactly one character”, you mean the same, repeated characted, then yes it is, though it is vanishingly unlikely.

As `crypto/rand` uses OS-provided randomness (either a syscall like `getentropy(3)` or `/dev/urandom`), if it provides you with non-random numbers, you have a bigger problem.

Reply:
There's still the case where it could avoid required characters, like all uppercase, or required symbols.

Reply:
Yeah, I called this out as well, that was what I suspect the reasoning for the division of the character sets previously.